zenoh/src/net/routing/interceptor/authorization.rs

```diff
@@ -0,0 +1,172 @@
+use ahash::RandomState;
+use std::collections::HashMap;
+use zenoh_config::{
+    AclConfig, Action, ConfigRule, Permission, PolicyRule, Subject, NUMBER_OF_ACTIONS,
+    NUMBER_OF_PERMISSIONS,
+};
+use zenoh_keyexpr::keyexpr;
+use zenoh_keyexpr::keyexpr_tree::{IKeyExprTree, IKeyExprTreeMut, KeBoxTree};
+use zenoh_result::ZResult;
+
+pub struct PolicyForSubject(Vec<Vec<KeTreeRule>>); //vec of actions over vec of permission for tree of ke for this
+pub struct PolicyMap(pub HashMap<i32, PolicyForSubject, RandomState>); //index of subject (i32) instead of subject (String)
+
+type KeTreeRule = KeBoxTree<bool>;
+
+pub struct PolicyEnforcer {
+    pub(crate) acl_enabled: bool,
+    pub(crate) default_permission: Permission,
+    pub(crate) subject_map: Option<HashMap<Subject, i32, RandomState>>,
+    pub(crate) policy_map: Option<PolicyMap>,
```

Comment:
Same comment, What's the reason of using an `Option<HashMap<_>>` and not simply an `HashMap`?